csharp/ql/src/utils/model-generator/CaptureNegativeSummaryModels.ql

```diff
@@ -0,0 +1,15 @@
+/**
+ * @name Capture negative summary models.
+ * @description Finds negative summary models to be used by other queries.
+ * @kind diagnostic
+ * @id cs/utils/model-generator/negative-summary-models
+ * @tags model-generator
+ */
+
+private import semmle.code.csharp.dataflow.ExternalFlow
+private import internal.CaptureModels
+private import internal.CaptureSummaryFlow
+
+from TargetApi api, string noflow
+where noflow = captureNoFlow(api) and not hasSummary(api, false)
```

Comment:
Why is the `not hasSummary(api, false)` bit needed?

Reply:
The CaptureFlow predicate might return a less specialized version of a handwritten summary or no summary at all (if flow was not identified programmatically). To avoid creating less specialized versions of the summaries or create negative summaries, when there might exist a handwritten one this has been made explicit here.
One could argue, whether we care at all about "conflicts" because at least for positive summaries we discard the less specialized version (due to the heuristic of trusting handwritten summaries over generated ones).
Should we remove the hasSummary(...) check completely? (I would like to do it as a follow up and include a re-gen of the models if we decide to do so).

Reply:
Thanks for the reminder; let's keep it in.